lnpbp-0001.md

```diff
@@ -188,102 +189,116 @@ protocol.
 all used public keys, so it can represent a correct input for the protocol.
 
-The tweaked procedure may result in public key that may, or may not have its *y*
+The tweaked procedure may result in a public key that may, or may not have its *y*
 coordinate being a quadratic residue (in terms of BIP-340 [4]). This may present
 a compatibility issue for using this scheme in Taproot/Schnorr-enabled outputs
 and protocols. Nevertheless, this issue may be mitigated by running the
-procedure  second time and replacing the original public key with its own
-negation if the resulted tweaked version  was not square.
+procedure a second time and replacing the original public key with its own
+negation, if the resulting tweaked version was not square.
 
-Proposal relies on a tagged hash prefix similar to the one used in BIP-340, [4],
+The proposal relies on a tagged hash prefix similar to the one used in BIP-340, [4],
 which helps to prevent protocol collisions.
 
 
 ## Rationale
 
-### Commitments with sets of public key, not a single key
+### Commitments with a set of public keys, not a single key
 
-The protocol was designed to support commitments to the multiple public key in
+The protocol was designed to support commitments to multiple public keys in
 order to be usable with non-P2(W)PK outputs. For instance, with Lightning 
 network all outputs in the commitment transaction are non-P2WPK, so all existing
 key tweaking schemes are not usable within LN structure.
 
 ### Use of HMAC insead of simple hash
 
-Reasons: 
-- prevention of length extension attacks;
-- makes impossible to forge a message and a public key equivalent to negation of
-  a tweaking factor multiplied to `G`, which will result in protocol failure.
+Reason 1: prevention of length-extension attacks
+
+As this protocol aims to be a generic scheme, the message `msg` can be of any length.
+If we would just use a simple hash (e.g. SHA256), users of `LNPBP-1` could **potentially** 
+be vulnerable to length-extension attacks, if they are not careful. To be on the safe side,
+we use HMAC-SHA256, which is resistant to length-extension attacks, but computationally
+more expensive. However, this protocol aims to be used in client-side validation applications 
+primarily and should therefore run many orders of magnitude less often then complete 
+validatation of all public blockchain data. The computational overhead of HMAC on a 
+client node is therefore considered negligible, for the targeted use cases.
+
+Reason 2: prevention of vocabulary attacks
+
+In certain circumstances a simple hash based commitment might be vulnerable to brute force
+vocabulary attacks, if the syntax and semantics of the invoking protokol are known to the attacker. 
+This is usally countered with adding additional entropy to each hash. In our case the public key
+`S` already provides enought entropy, which - when added via HMAC-SHA256 to the whole `msg` -
+sufficiently counters such vocabulary attacks.
```

Comment:
The same applies when just a hash is used, since we serialize the public key into the hashing function, adding its entropy with the same efficiency as with HMAC. Nevertheless the cause is well described and I just propose to move it into a "No nonce" section in Rationale, merging it with a text given there.

Corrected some typos:
```suggestion
Reason 2: prevention of vocabulary attacks

In certain circumstances a simple hash based commitment might be vulnerable to brute force
vocabulary attacks, if the syntax and semantics of the invoking protocol are known to the attacker. 
This is usually countered with adding additional entropy to each hash. In our case the public key
`S` already provides enough entropy, which - when added via HMAC-SHA256 to the whole `msg` -
sufficiently counters such vocabulary attacks.
```